docs/specification/draft/basic/transports.md

```diff
@@ -48,38 +48,209 @@ sequenceDiagram
     deactivate Server Process
 ```
 
-## HTTP with SSE
+## Streamable HTTP
 
-In the **SSE** transport, the server operates as an independent process that can handle
-multiple client connections.
+{{< callout type="info" >}} This replaces the [HTTP+SSE
+transport]({{< ref "/specification/2024-11-05/basic/transports#http-with-sse" >}}) from
+protocol version 2024-11-05. See the [backwards compatibility](#backwards-compatibility)
+guide below. {{< /callout >}}
 
-The server **MUST** provide two endpoints:
+In the **Streamable HTTP** transport, the server operates as an independent process that
+can handle multiple client connections. This transport uses standard HTTP with optional
+[Server-Sent Events](https://en.wikipedia.org/wiki/Server-sent_events) (SSE) for
+streaming multiple server messages. This permits "plain HTTP" MCP servers, as well as
+more feature-rich servers supporting streaming and server-to-client notifications and
+requests.
 
-1. An SSE endpoint, for clients to establish a connection and receive messages from the
-   server
-2. A regular HTTP POST endpoint for clients to send messages to the server
+The server **MUST** provide a single HTTP endpoint path (hereafter referred to as the
+**MCP endpoint**) that supports both POST and GET methods. For example, this could be a
+URL like `https://example.com/mcp`.
 
-When a client connects, the server **MUST** send an `endpoint` event containing a URI for
-the client to use for sending messages. All subsequent client messages **MUST** be sent
-as HTTP POST requests to this endpoint.
+### Message Exchange
 
-Server messages are sent as SSE `message` events, with the message content encoded as
-JSON in the event data.
+1. Every JSON-RPC message sent from the client **MUST** be a new HTTP POST request to the
+   MCP endpoint.
+
+2. When the client sends a JSON-RPC _request_ to the MCP endpoint via POST:
+
+   - The client **MUST** include an `Accept` header, listing both `application/json` and
```

Comment:
Is this going to be trick in clients? E.g. how would you use the [httpx SSE](https://pypi.org/project/httpx-sse/) package when you don't yet know if you're going to get back an SSE stream?

Maybe this isn't that hard, just checking.

Reply:
The client would look at the header to know what it is receiving

Reply:
(SSE itself is incredibly simple in implementation too, so it wouldn't be too hard to roll something custom as necessary.)

Reply:
Actively maintained libraries would adjust to support the functionality -- or someone from this community would provide a patch.

Reply:
Would it be worth providing an example implementation in the spec to clarify how clients are expected to support this, or that existing clients assuming sse neee to be modified? It also confused me a bit how this would work